test/sqllogictest/advent-of-code/2023/aoc_1217.slt

```diff
@@ -312,7 +312,7 @@ Explained Query:
       cte l7 =
         Reduce aggregates=[min(#0)]
           Project (#2)
-            Join on=(#0 = #3 AND #1 = #4) type=differential
+            Join on=(#0 = #3 AND #1 = #4) type=delta
```

Comment:
Probably good change: intermediate results don't need to be arranged.

Reply:
Same as above, reverted to a differential join after I [tweaked the heuristics](https://github.com/MaterializeInc/materialize/pull/31281#discussion_r1943159959).